src/overlays/actors/ovl_En_Door/z_en_door.c

```diff
@@ -27,29 +39,328 @@ const ActorInit En_Door_InitVars = {
+    if (DECR(this->lockTimer) == 0) { // Wait for the lock to finish if applicable
+        if (SkelAnime_FrameUpdateMatrix(&this->skelAnime) != 0) {
+            // If done, return to the idle action and free the player
+            this->actionFunc = EnDoor_Idle;
+            this->unk_191 = 0;
+        } else if (func_800A56C8(&this->skelAnime, sDoorAnimOpenFrames[this->unk_190]) != 0) {
+            // Play the appropriate sound on open
+            Audio_PlayActorSound2(&this->actor, (globalCtx->sceneNum == SCENE_HAKADAN || 
+                globalCtx->sceneNum == SCENE_HAKADANCH || 
+                globalCtx->sceneNum == SCENE_HIDAN) ? NA_SE_EV_IRON_DOOR_OPEN : NA_SE_OC_DOOR_OPEN);
+            if (this->skelAnime.animPlaybackSpeed < 1.5f) {
+                numEffects = (s32) (Math_Rand_ZeroOne() * 30.0f) + 50;
+                for (i = 0; i < numEffects; i++) {
+                    func_800293E4(globalCtx, &this->actor.posRot.pos, 60.0f, 100.0f, 50.0f, 0.15f);
+                }
+            }
+        } else if (func_800A56C8(&this->skelAnime, sDoorAnimCloseFrames[this->unk_190]) != 0) {
+            // Play the appropriate sound on close
+            Audio_PlayActorSound2(&this->actor, (globalCtx->sceneNum == SCENE_HAKADAN || 
+                globalCtx->sceneNum == SCENE_HAKADANCH || 
+                globalCtx->sceneNum == SCENE_HIDAN) ? NA_SE_EV_IRON_DOOR_CLOSE : NA_SE_EV_DOOR_CLOSE);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Door/func_809FC8F4.s")
+void EnDoor_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnDoor* this = THIS;
+    this->actionFunc(this, globalCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Door/func_809FC928.s")
+s32 EnDoor_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    s32 pad;
+    TransitionActorEntry* transitionEntry;
+    Gfx** temp_a2;
+    s32 pad2;
+    s16 phi_v0_2;
+    s32 phi_v0;
+    EnDoor* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Door/func_809FC998.s")
+    if (limbIndex == 4) {
+        temp_a2 = &D_809FCEE4[(s8)this->displaylistIdx];
+        transitionEntry = &globalCtx->transitionActorList[(u16)this->actor.params >> 0xA];
+        rot->z += this->actor.posRot.rot.y;
+        if ((globalCtx->roomCtx.prevRoom.num >= 0) || (transitionEntry->sides[0].room == transitionEntry->sides[1].room)) {
+            phi_v0_2 = ((this->actor.shape.rot.y + this->skelAnime.limbDrawTbl[3].z) + rot->z) - Math_Vec3f_Yaw(&globalCtx->view.eye, &this->actor.posRot.pos);
+            *dList = (ABS(phi_v0_2) < 0x4000) ? temp_a2[0] : temp_a2[1];
+        } else {
+            phi_v0 = this->unk_192;
+            if (transitionEntry->sides[0].room != this->actor.room) {
+                phi_v0 ^= 1;
+            }
+            *dList = temp_a2[phi_v0];
+        }
+    }
+    return 0;
```

Comment:
```suggestion
    return false;
```
the return value is a "skipDrawingLimb" boolean